### PR TITLE
feat: add top-level error boundary

### LINKED ISSUE #1700: Improve UX of app crash
**Is your feature request related to a problem? Please describe.**
When the application crashes I see a black screen and I must restart the application.

**Describe the solution you'd like**
I'd like a warm animation and a button that takes me home. Maybe we could include a link to our GitHub to make it easier to report an issue.

### PR DESCRIPTION
**Pull request by WitoDelnat**
This PR closes #1700 and improves the experience when a user runs into a crash.

## Changes

- Adds an error boundary and error page.
- Allows easily reporting the bug.
- Allows to go back to project selection without having to restart Monokle.

## How to test it

- Add an error that would crash the application.
- Trigger the error.

## Checklist

- [x] tested locally
- [x] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Comment by WitoDelnat** [comment]
Maybe we should add a third action? "Talk with us" and redirect towards our Discord for direct line of support?

File: src/components/organisms/ErrorPage/ErrorPage.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+export const ErrorPage: React.FC<FallbackProps> = ({error, resetErrorBoundary}) => {
+  const [showError, setShowError] = useState<boolean>(false);
+
+  const createGitHubIssue = useCallback(() => {
+    const url = newGithubIssueUrl({
+      user: 'kubeshop',
+      repo: 'monokle',
+      title: '[crash] Something went wrong',
+      body: `**Describe the bug**\n\n\n**Steps to reproduce**\n\n\n**Stacktrace** \n\n \`\`\`\n${error.message}\n ${error.stack}\n\`\`\``,
```
Comment (WitoDelnat) [inline src/components/organisms/ErrorPage/ErrorPage.tsx:23]: We should add the deviceId here so we can track our analytics, etc with it.

**Comment by WitoDelnat** [comment]
How to test: you'll have to quickly force a button to throw an error. 

```
// src/components/molecules/ResourceFilter/ResourceFilter.tsx:169
// This throws an error when you open the resource filter in the navigator.
useEffect(() => {
  throw new Error('BOOM');

  // if (!wasLocalUpdate) {
  //   setName(filtersMap.name);
  //   setKind(filtersMap.kind);
  //   setNamespace(filtersMap.namespace);
  //   setLabels(filtersMap.labels);
  //   setAnnotations(filtersMap.annotations);
  //   setFileOrFolderContainedIn(filtersMap.fileOrFolderContainedIn);
  // }
}, [wasLocalUpdate, filtersMap]);
```

Updated style video:

https://user-images.githubusercontent.com/7761005/167799847-1e881d0a-a598-499b-b119-df1004535a2e.mov

File: src/components/organisms/StartProjectPane/RecentProjectsPage.tsx [hunk 2]
```diff
@@ -42,6 +42,7 @@ const NewRecentProjectsPane = () => {
     dispatch(toggleProjectPin(project));
   };
 
+  console.log('render recents');
```
Comment (topliceanurazvan) [inline src/components/organisms/StartProjectPane/RecentProjectsPage.tsx:45]: Is this needed?

File: src/redux/store.ts [hunk 3]
```diff
@@ -24,17 +24,31 @@ if (process.env.NODE_ENV === `development`) {
...
+  navigator: navigatorSlice.reducer,
+  uiCoach: uiCoachSlice.reducer,
+  extension: extensionSlice.reducer,
+});
+
+const rootReducer: typeof appReducer = (state, action) => {
+  if (resetStore.match(action)) {
+    // Invoking reducers with `undefined` sets initial state.
+    // see https://stackoverflow.com/questions/35622588/how-to-reset-the-state-of-a-redux-store/35641992#35641992
+    return appReducer(undefined, action);
```
Comment (topliceanurazvan) [inline src/redux/store.ts:44]: Nice approach here 👍🏻

**Review by topliceanurazvan — APPROVED** [review]
LGTM!

File: src/components/organisms/StartProjectPane/RecentProjectsPage.tsx (hunk 2 above)
Comment (WitoDelnat) [reply src/components/organisms/StartProjectPane/RecentProjectsPage.tsx:45]: Good catch!

**Pull request merged**